examples/changelog_generator/jira/.changelog-generator.sh

```diff
@@ -0,0 +1,159 @@
+## This is and example of changelog generator for projects that use JIRA.
+## Script calls a python script that uses JIRA package to fetch issues from JIRA API
+## Generated changelog is in format
+##
+## * [<issue number> <issue title>](<link to issue>)
+## * [<issue number> <issue title>](<link to issue>)
+## ...
+##
+## depending on how many issue numbers are provided.
+## Script can also be configured to parse the task number for the branch name.
+
+
+## In order to use this script you'll need to follow these steps
+##
+## 1. Install JIRA package ("pip install jira" will do the trick)
+## 2. Create a JIRA token (https://id.atlassian.com/manage-profile/security/api-tokens)
+## 3. Open .zshrc (or .bash_profile)
+## 4. Add these two lines:
+##      export JIRA_EMAIL="<your email on jira>"
+##      export JIRA_TOKEN="<your jira token>"
+## 5. Save and close .zshrc (or .bash_profile) and restart terminal (or run source ~/.zshrc or source ~/.bash_profile)
+##
+## Now you should be all set to use this script in your app-deploy.
+
+
+# Regex used to find task numbers in branch name.
+# Replace <project> with prefix used in task numbers on your project.
+# Set to "" if you don't want to use this feature.
+BRANCH_NAME_TASK_MATCHING_REGEX="(<project>-[0-9]+)"
+
+# Path to python script that will fetch issues from JIRA.
+PYTHON_SCRIPT_PATH=".changelog-generator.py"
+
+# JIRA project URL. Used by python script to construct a url to specific issue.
+# Replace <project> with your JIRA project name.
+JIRA_PROJECT_URL="https://<project>.atlassian.net"
+
+function generate_changelog {
+
+    if [[ -z "$JIRA_EMAIL" || -z "$JIRA_TOKEN" ]]; then
+        echo "Missing JIRA_EMAIL or JIRA_TOKEN. Please add it to your .zshrc or .bash_profile configuration file."
+        echo
+
+        return
+    fi
+
+    local task_numbers=""
+
+    if [[ ! -z $BRANCH_NAME_TASK_MATCHING_REGEX ]]; then
+        # Task number contained in branch name
+        current_branch=`git rev-parse --abbrev-ref HEAD`
+        if [[ $current_branch =~ $BRANCH_NAME_TASK_MATCHING_REGEX ]]; then
+            task_numbers=$(printf "%s " "${BASH_REMATCH[@]}")
```

Comment:
Be careful with `printf`. Not all bash versions support it. I think I had issues with it, so I tried to avoid using it.